alerts/machine-config-operator/MachineConfigControllerPausedPoolKubeletCA.md

```diff
@@ -0,0 +1,165 @@
+
+### Long term: critical
+
+The `kube-apiserver-to-kubelet-signer` certificate in the `kubelet-ca.crt`
+bundle is what enables trust between your node's kubelet and your cluster. The
+apiserver schedules a rotation when this certificate reaches 80% of its
+(currently 365 day) lifetime. The new one *must* be deployed before all of the
+previous ones expire.
+
+The kubelets on the nodes in the specified pool will stop communicating with the
+cluster apiserver if the `kube-apiserver-to-kubelet-signer`  is allowed to
+expire. The nodes will no longer be able to participate in the cluster. *This is
+very bad*.
+
+To avoid this, you will need to unpause the specified pool to allow the new
+`kubelet-ca.crt` bundle to be deployed.
+
+## Diagnosis
+
+If a pool is paused, it was paused by an administrator; pools do not pause
+themselves automatically.
+
+>NOTE: there are some operators (like SR-IOV) that may briefly pause a pool to
+>do some work, but they will not leave the pool paused long-term.
+
+For the commands below, replace `$MCP_NAME` with the name of your pool.
+
+You can see the pool's paused status by looking at the `spec.paused` field for
+the pool:
+
+```console
+oc -n openshift-machine-config-operator get mcp $MCP_NAME -o jsonpath='{.spec.paused}'
+```
+
+For the `kube-apiserver-to-kubelet-signer` certificate in the cluster, you can
+check its annotations to see when it was rotated:
+
+```console
+oc -n openshift-kube-apiserver-operator describe secret kube-apiserver-to-kubelet-signer
+```
+
+The `kubelet-client` cert (the one that is responsible for `oc logs`, etc
+working) in the cluster, you can check which signer it was signed with:
+
+```console
+oc -n openshift-kube-apiserver describe secret/kubelet-client
+```
+
+You can also check the certificates that are present in the bundle on one of
+your nodes to see when they expire:
+
+```console
+openssl crl2pkcs7 -nocrl -certfile /etc/kubernetes/kubelet-ca.crt | openssl pkcs7 -print_certs -text -noout
+``````
+
+>NOTE: There may be multiple `kube-apiserver-to-kubelet-signer` certificates in
+>the certificate bundles, kube-apiserver does not pull them out until they
+>expire. You want to look at the *newest* `kube-apiserver-to-kubelet-signer`.
+
+You can find which MachineConfig a pool is currently using by looking at the
+pool status:
+
+```console
+oc -n openshift-machine-config-operator get mcp $MCP_NAME -o jsonpath='{.status.configuration.name}'
+```
+
+Checking the expiry dates in the MachineConfig yourself is less pleasant because
+the bundle is encoded.
+
+You can use something like this to decode it if it's urlencoded:
+
+```console
+oc get mc rendered-worker-bc1470f2331a3136999e0b49d85e1e21 -o jsonpath='{.spec.config.storage.files[?(@.path=="/etc/kubernetes/kubelet-ca.crt")].contents.source}' | python3 -c 'import sys, urllib.parse; print(urllib.parse.unquote(sys.stdin.read()))' | openssl x509 -text -noout
+```
+
+Or something like this it if is base64 + gzipped:
```

Comment:
If the bundle is base64-encoded and gzipped, use the following command with the desired MC to decode it: